song-server/src/main/java/bio/overture/song/server/service/AnalysisService.java

```diff
@@ -134,21 +129,18 @@
 
   @Transactional
   public String create(
-      @NonNull String studyId, @NonNull Payload payload, boolean ignoreAnalysisIdCollisions) {
+      @NonNull String studyId, @NonNull Payload payload) {
     studyService.checkStudyExist(studyId);
-    val inputAnalysisId = payload.getAnalysisId();
 
-    // This doesnt commit the id to the id server
-    val candidateAnalysisId =
-        resolveCandidateAnalysisId(inputAnalysisId, ignoreAnalysisIdCollisions);
+    val analysisId = idService.generateAnalysisId();
 
     // Prevent duplicate analyses from being created
     checkServer(
-        !isAnalysisExist(candidateAnalysisId),
+        !isAnalysisExist(analysisId),
```

Comment:
In the (very unlikely) even that there is a random number collision, there's nothing the user can do about it.
Either remove the check, or, if you're paranoid enough, wrap it in a while(collision) { generate_new_random_number()} type loop.